js.go

```diff
@@ -2861,7 +2861,13 @@ func (sub *Subscription) Fetch(batch int, opts ...PullOpt) ([]*Msg, error) {
 	}
 	var hbTimer *time.Timer
 	var hbErr error
-	if err == nil && len(msgs) < batch {
+	sub.mu.Lock()
```

Comment:
Maybe extract following logic
```go 
	sub.mu.Lock()
	subClosed := sub.closed || sub.draining
	sub.mu.Unlock()
```

in a new function inside Subscription e.g. `isClosedOrDraining()` or similar

and reuse it in both places: `Fetch` and `FetchBatch`?

Reply:
That's such a tiny piece of logic only used in 2 places, I would honestly just leave it inlined for clarity :)

With those kinds of checks I usually wait until I have to repeat it at least 3 times before extracting to a separate method, but that's just preference.